src/amo/components/AddonReviewCard/index.js

```diff
@@ -298,25 +359,25 @@ export class AddonReviewListItemBase extends React.Component<InternalProps> {
         siteUser &&
         !replyingToReview &&
         !review.reply &&
-        !isReply &&
+        !this.isReply() &&
         (isAddonAuthor({ addon, userId: siteUser.id }) ||
           siteUserHasReplyPerm) &&
         review.userId !== siteUser.id ? (
           <a
             href="#reply"
             onClick={this.onClickToBeginReviewReply}
-            className="AddonReviewListItem-begin-reply AddonReviewListItem-control"
+            className="AddonReviewCard-begin-reply AddonReviewCard-control"
           >
             <Icon name="reply-arrow" />
             {i18n.gettext('Reply to this review')}
           </a>
         ) : null}
 
-        {review ? (
+        {flaggable && review ? (
```

Comment:
We've had that feature for a while and it looks like this:

<img width="927" alt="screenshot 2018-09-07 09 41 53" src="https://user-images.githubusercontent.com/55398/45225495-4d77f580-b282-11e8-8f1c-2235b8c70e08.png">

There are different schools of thought here. I find it really confusing when menu items (or top level controls) are hidden for different modes. For example, I'd be looking for a control and scratching my head wondering where it is. I prefer that the *control* is still there but it's either disabled or non-functional.

However, rendering a saved review is a special case so I think it makes sense to remove the *flag* control entirely. Since it's not part of a list of similar components, one would not expect the same consistency. Also, the layout is crowded with three control links. It looks better with only two (*edit* and *delete*).

Reply:
I guess we'll just have to agree to disagree on this one. While I see your point about a menu item disappearing, the fact that a user can click on it, and then be told "you really shouldn't have clicked on this, because it doesn't do anything" is equally, if not more, frustrating, imo.